src/tasks/07_groupByType.js

```diff
@@ -21,3 +21,18 @@
  */
 
 // <-- тут реализация groupByType
+export function groupByType(array) {
+    let result = {
+        legal: [],
+        physical: []
+    };
+    array.forEach(element => {
+        if (element.type === 'physical') {
+            result.physical.push({name: element.name});
+        }
+        else {
+             result.legal.push({name: element.name});
+        }
+    });
+    return result
```

Comment:
Для таких операций можно использовать reduce
```suggestion
    return array.reduce((result, element) => {
        if (element.type === 'physical') {
            result.physical.push({name: element.name});
        }
        else {
             result.legal.push({name: element.name});
        }
        return result;
    }, {
        legal: [],
        physical: []
    });
```